src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -860,7 +860,8 @@ private void GetDisposalInfoForEnumerator(ref ForEachEnumeratorInfo.Builder buil
             {
                 builder.NeedsDisposal = true;
             }
-            else if (enumeratorType.IsRefLikeType || isAsync)
+            else if (Compilation.LanguageVersion >= MessageID.IDS_FeatureUsingDeclarations.RequiredVersion() &&
```

Comment:
>Compilation.LanguageVersion >= MessageID.IDS_FeatureUsingDeclarations.RequiredVersion() [](start = 21, length = 87)

Nit: we could just use `Compilation.IsFeatureEnabled`

Reply:
Yeah, please use the helper here, because when we sometimes change these semantics to have an extra requirement it's easier to change the helper than to find all the strict integer comparisons.

Reply:
IDK that existed, much neater :)